posthog/src/main/java/com/posthog/internal/PersonProfileMode.kt

```diff
@@ -0,0 +1,22 @@
+package com.posthog.internal
+
+public enum class PersonProfileMode(public val mode: String) {
+    NEVER("never"),
+    ALWAYS("always"),
+    IDENTIFIED_ONLY("identified_only"),
+    ;
+
+    public companion object {
+        public fun fromConfig(config: String?): PersonProfileMode {
+            return values().find { it.mode == config } ?: ALWAYS
+        }
+    }
+
+    public fun shouldProcessProfile(event: String): Boolean {
+        return when (this) {
+            NEVER -> false
+            ALWAYS -> true
+            IDENTIFIED_ONLY -> event == "\$identify" || event == "\$create_alias" || event == "\$groupidentify"
```

Comment:
`_hasPersonProcessing` does not have those checks in JS, or did I miss something?

Reply:
This should check a few other things apparently such as https://github.com/PostHog/posthog-js/blob/34204f4aff107162bdb259d2a7c855d67bf0260c/src/posthog-core.ts#L1905-L1906